lib/spack/spack/package.py

```diff
@@ -1257,6 +1258,16 @@ def _update_explicit_entry_in_db(self, rec, explicit):
                 message = '{s.name}@{s.version} : marking the package explicit'
                 tty.msg(message.format(s=self))
 
+    def try_install_from_binary_cache(self, explicit):
+        specs = binary_distribution.get_specs()
+        if self.spec not in specs:
+            return False
+        tarball = binary_distribution.download_tarball(self.spec)
```

Comment:
Can we add a tty message here "Installing <pkg> from binary cache" or "Installing <pkg> from mirror"? Something that effectively replaces the normal install output and informs the user what's going on under the hood when they install packages.

Maybe a message before we try the binary installation "Searching for binary cache of <pkg>" and then a message that either says "Installing <pkg> from source" if we have no binary cache for it or "Installing <pkg> binary" if we do have one.

I'm not sure that I have the exact answer on implementation or exactly which messages we should provide, but the general idea is that we should provide equally robust output for binary cache installation as for installation from source.